Fix support for Atom 1.37 (and maybe a few older versions)

rixo opened the pull request:
I've recently upgraded to Atom 1.37 (from 1.26) and it broke recent files finder.

The cause is a mandatory arg that has recently been added to the constructor of FuzzyFinderView you're extending: https://github.com/atom/fuzzy-finder/commit/6fbfe3587497ebb3a4f8262e0d9334975a5f40b6#diff-e0ee5d9b066d4ea2a281ddaf5be6993eR26

I looked into fuzzy-finder's code to see how it gets its `metricsReporter` and it turns out it is a service it consumes. So I could replicate into this project and it fixed my error.

I'm not too sure that my fix is semantically correct since I don't know anything about metricsReporter. And I'm not too strong on the Coffee side. But I'd really like recent files finder to continue working in recent Atoms, so here's a PR that you can at least use for reference.

rixo committed:
Fix support for Atom 1.37 (and maybe a few older versions)

rixo committed:
fix runtime errors & lint error

viddo commented on lib/main.coffee:
```diff
@@ -33,9 +35,21 @@ module.exports =
+      @recentFilesView = new RecentFilesView(@getMetricsReporter(), @recentFiles)
     @recentFilesView
 
+  getMetricsReporter: ->
```
Considering we do not care about these metrics, and do not want to pollute stats for FuzzyFinder, I'd prefer don't even use their implementation. Instead, I suggest we create a proxy object that makes any function calls no-ops.

```suggestion
  getMetricsReporter: ->
    new Proxy(new Function, {
      get: (target, prop, receiver) -> receiver,
    })
```

viddo reviewed (changes requested):
Hi there!
Ouch, good catch! I guess I have not had to use my this in a while myself so had not noticed.

In addition to the inline suggestion (re: `getMetricsReporter`), can you remove the `consumeMetricsReporter` function and package.json entry? I think it's better if this plugin don't pollute their metrics with wrong data, and I don't think we're required to consume that service, right?

Thanks for the pull-request!

rixo committed:
Use inert metrics proxy to prevent pollution of fuzzy-finder metrics

rixo committed:
Oops. Forgot to actually use the newly added metrics proxy.

rixo committed:
Remove reference to dummy metrics proxy except where it is really used

rixo committed:
Remove unused import

rixo commented:
I must confess I fail to appreciate the ins and outs of these metrics... But you must be right, we can't do harm if we do nothing.

So I added a noop proxy and cleaned things a bit following this change.

Sorry for butchering the commit history.

viddo commented:
No worries, thanks!
